Bottom: 57c49bb54c28ffd41da0b641c52c54a95e397ab5
Top:    46443fdac177f3a6d9a4cd66b4e645ecf3cd501c
Author: Ian Kent <raven@themaw.net>
Date:   2013-05-11 09:33:30 +0800

autofs-5.0.7 - make dump maps check for duplicate indirect mounts


---

diff --git a/lib/master.c b/lib/master.c
index 0cf9de8..9e2ecfb 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1313,6 +1313,7 @@ int master_show_mounts(struct master *master)
 		struct master_mapent *this;
 		struct autofs_point *ap;
 		time_t now = time(NULL);
+		unsigned int count = 0;
 		int i;
 
 		this = list_entry(p, struct master_mapent, list);
@@ -1363,6 +1364,9 @@ int master_show_mounts(struct master *master)
 					printf("  map: %s\n", source->argv[0]);
 					i = 1;
 				}
+				if (count && ap->type == LKP_INDIRECT)
+					printf("  duplicate map source will be"
+						" ignored at run time\n");
 				if (source->argc > 1) {
 					printf("  arguments: ");
 					for (; i < source->argc; i++)
@@ -1382,6 +1386,8 @@ int master_show_mounts(struct master *master)
 				} while ((me = cache_lookup_next(source->mc, me)));
 			}
 
+			count++;
+
 			source = source->next;
 		}
